eval.py

```diff
@@ -1409,12 +1411,12 @@ def main():
         args.trained_model = SavePath.get_latest('weights/', cfg.name)
     elif is_valid_stub(args.trained_model):
         if args.engine == 'torch':
-            args.trained_model = get_checkpoint_from_stub(args.trained_model)
+            trained_model_weights = get_checkpoint_from_stub(args.trained_model)
         else:
-            args.trained_model = get_model_onnx_from_stub(args.trained_model)
+            trained_model_weights = get_model_onnx_from_stub(args.trained_model)
 
     args.engine = get_engine(
-        model_filepath=args.trained_model,
+        model_filepath=trained_model_weights,
```

Comment:
`trained_model_weights` can be undefined at this point based on conditional block above. Same applies to references below

Reply:
Very good point, cooking up a fix.